spec/latest/json-ld-framing/index.html

```diff
@@ -552,34 +543,109 @@
+      the values for <code>@value</code>, <code>@type</code>, and <code>@language</code>
+      may also be an array of one or more <a>string</a> values.</dd>
+  </dl>
+
+<p>The frame matching algorithm takes the <a>framing state</a> (<em>state</em>),
+  a list of subjects to match from the <a>map of flattened subjects</a> (<em>subjects</em>),
+  a <a>frame</a> to match against (<em>frame</em>), and the <em>requireAll</em> flag
+  and returns a list of matched subjects by filtering each <em>node</em> in <em>subjects</em> as follows:</p>
+
+<p>Frame matching follows a cascade, first attempting to match on a particular <code>@id</code>, then
+  a particular <code>@type</code> (or lack of <code>@type</code>), then by matching on any or all
+  of a set of properties, if neither <code>@id</code>, nor <code>@type</code> are in the frame.</p>
+
+<ol class="algorithm">
+  <li><em>Node</em> matches if it has an <code>@id</code> property including any <a>IRI</a> in the <code>@id</code> property in <em>frame</em>.</li>
+  <li><em>Node</em> matches if frame has no non-<code>keyword</code> properties.</li>
+  <li>If <em>requireAll</em> is <strong>true</strong>, <em>node</em> matches if all non-<a>keyword</a> properties (<em>property</em>) in <em>frame</em> match any of the following conditions.
+    Or, if <em>requireAll</em> is <strong>false</strong>, if any of the non-<a>keyword</a> properties (<em>property</em>) in <em>frame</em> match any of the following conditions.
+    For the <em>values</em> of each <em>property</em> from <em>frame</em> in <em>node</em>:
+    <ol class="algorithm">
+      <li>If <em>property</em> is <code>@type</code>:
+        <ol class="algorithm">
+          <li><em>Property</em> matches if the <code>@type</code> property in frame includes any <a>IRI</a> in <em>values</em>.</li>
+          <li>Otherwise, <em>property</em> matches if <em>values</em> is not empty and the <code>@type</code> property in <em>frame</em> is <code><a>wildcard</a></code>.</li>
+          <li>Otherwise, <em>property</em> matches if <em>values</em> is empty and the <code>@type</code> property in <em>frame</em> is <code><a>match none</a></code>.</li>
+          <li>Otherwise, <em>property</em> does not match.</li>
+        </ol>
+      </li>
+      <li>Otherwise, the value of <em>property</em> in <em>frame</em> MUST be empty, or an array
+        containing a valid <a>frame</a>.</li>
+      <li><em>Property</em> matches if <em>values</em> is empty, or non existant,
+        the value of <em>property</em> in <em>frame</em>
+        is a <a>dictionary</a> containing only the key <code>@default</code> with any value,
+        and any other property in <em>node</em> has a non-default match.</li>
+      <li><em>Node</em> does not match if <em>values</em> is not empty and the value of <em>property</em> in <em>frame</em> is <code><a>match none</a></code>, and further matching is aborted.</li>
```

Comment:
The change is because we match a specific property rather than the whole node. If `requireAll` is set, all properties must have matched. Otherwise, any property. A mis-matching property does not necessarily cause the node match to fail.